src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -267,44 +290,74 @@ def onMouseClickOnImage(self, x: float, y: float):
             radial_value + self._delta_radial_over_2,
         )
 
-    def getColorLastCurve(self):
-        curves = self._integrated_plot_widget.getAllCurves()
-        if curves:
-            return self._integrated_plot_widget.getAllCurves()[-1].getColor()
+    def getColorCurve(self, legend: str):
```

Comment:
```suggestion
    def getCurveColor(self, legend: str):
```
The actual thing you are retrieving is the curve color (or the color of the curve), not a curve.